src/internal.c

```diff
@@ -18056,7 +18081,7 @@ int DecodePrivateKey(WOLFSSL *ssl, word16* length)
             if ((*inOutIdx - begin) + len > size)
                 return BUFFER_ERROR;
 
-            PickHashSigAlgo(ssl, input + *inOutIdx, len);
+            (void)PickHashSigAlgo(ssl, input + *inOutIdx, len);
```

Comment:
If an error is now possible why aren't we checking it?

Reply:
Checking return now.
Only need to check return when there is a certificate and key to use.
List of algorithms pairs didn't include ECC algorithms if ECC certificate with RSA signature. But then failed when client wanted to use ECC certificate with ECDSA certificate. Therefore signature algorithms include ECC if haveECDSA OR haveECC and similarly for RSA.